### PR TITLE
Add `proximity_placement_group_id` to AKS node pools

### LINKED ISSUE #9164: Support for proxiomity groups in azurerm_kubernetes_cluster
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Based on referenced document it should be possible to set proximity group on node pools created for AKS. It would be nice to have this ability in provider.

### New or Affected Resource(s)

* azurerm_kubernetes_cluster
* azurerm_kubernetes_cluster_node_pool

### Potential Terraform Configuration

```hcl
resource "azurerm_kubernetes_cluster" "example" {
  name                = "example-aks1"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  dns_prefix          = "exampleaks1"

  default_node_pool {
    name       = "default"
    node_count = 1
    vm_size    = "Standard_D2_v2"
    proximity_placement_group_id = "00000000-0000-0000-0000-000000000000"
  }

  service_principal {
    client_id     = "00000000-0000-0000-0000-000000000000"
    client_secret = "00000000000000000000000000000000"
  }
}
```

### References

* https://docs.microsoft.com/en-gb/azure/aks/reduce-latency-ppg

### PR DESCRIPTION
**Pull request by favoretti**
Fixes #9164

(make fmt reformatted images.html and
security_center_auto_provisioning.html markdown, although I made no
changes there)

### CODE REVIEW

File: azurerm/internal/services/containers/kubernetes_cluster_node_pool_data_source.go [hunk 1]
```diff
@@ -248,6 +253,14 @@ func dataSourceKubernetesClusterNodePoolRead(d *schema.ResourceData, meta interf
 		}
 		d.Set("priority", priority)
 
+		proximityPlacementGroupId := ""
+		if props.ProximityPlacementGroupID != nil {
+			proximityPlacementGroupId = string(*props.ProximityPlacementGroupID)
+		}
+		d.Set("proximity_placement_group_id", proximityPlacementGroupId)
+
+		d.Set("min_count", minCount)
```
Comment (tombuildsstuff) [inline azurerm/internal/services/containers/kubernetes_cluster_node_pool_data_source.go:262]: this is already set above: https://github.com/terraform-providers/terraform-provider-azurerm/pull/9195/files#diff-7ebfe4f8c2f575e50130c59761183244b759429db1dcce690bd9ef97c61cd7ecR219

```suggestion
```

File: azurerm/internal/services/containers/kubernetes_nodepool.go [hunk 2]
```diff
@@ -133,6 +133,11 @@ func SchemaDefaultNodePool() *schema.Schema {
 					Computed:     true,
 					ValidateFunc: validation.StringIsNotEmpty,
 				},
+				"proximity_placement_group_id": {
+					Type:         schema.TypeString,
+					Optional:     true,
+					ValidateFunc: validation.StringIsNotEmpty,
```
Comment (tombuildsstuff) [inline azurerm/internal/services/containers/kubernetes_nodepool.go:139]: the `compute/validate` package (`github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/compute/validate`) exposes a method of validating this is a ProximityPlacementGroupID - could we use that here:

```suggestion
					ValidateFunc: computeValidate.ProximityPlacementGroupID,
```

File: azurerm/internal/services/containers/kubernetes_nodepool.go [hunk 3]
```diff
@@ -133,6 +133,11 @@ func SchemaDefaultNodePool() *schema.Schema {
 					Computed:     true,
 					ValidateFunc: validation.StringIsNotEmpty,
 				},
+				"proximity_placement_group_id": {
+					Type:         schema.TypeString,
+					Optional:     true,
```
Comment (tombuildsstuff) [inline azurerm/internal/services/containers/kubernetes_nodepool.go:138]: the VMSS resources (which this is provisioning under the hood) requires this to be ForceNew, so I assume this does too?

```suggestion
					Optional:     true,
					ForceNew: true,
```

File: website/docs/d/kubernetes_cluster_node_pool.html.markdown [hunk 4]
```diff
@@ -70,6 +70,8 @@ In addition to the Arguments listed above - the following Attributes are exporte
 
 * `priority` - The priority of the Virtual Machines in the Virtual Machine Scale Set backing this Node Pool.
 
+* `proximity_placement_group_id` - ID of `azurerm_proximity_placement_group` where the Virtual Machine Scale Set backing this Node Pool will be placed.
```
Comment (tombuildsstuff) [inline website/docs/d/kubernetes_cluster_node_pool.html.markdown:73]: can we make this:

```suggestion
* `proximity_placement_group_id` - The ID of the Proximity Placement Group where the Virtual Machine Scale Set backing this Node Pool will be placed.
```

File: website/docs/r/kubernetes_cluster_node_pool.html.markdown [hunk 5]
```diff
@@ -99,6 +99,8 @@ The following arguments are supported:
 
 * `priority` - (Optional) The Priority for Virtual Machines within the Virtual Machine Scale Set that powers this Node Pool. Possible values are `Regular` and `Spot`. Defaults to `Regular`. Changing this forces a new resource to be created.
 
+* `proximity_placement_group_id` - (Optional) The ID of `azurerm_proximity_placement_group` where the Virtual Machine Scale Set that powers this Node Pool will be placed.
```
Comment (tombuildsstuff) [inline website/docs/r/kubernetes_cluster_node_pool.html.markdown:102]: can we make this:

```suggestion
* `proximity_placement_group_id` - (Optional) The ID of the Proximity Placement Group where the Virtual Machine Scale Set that powers this Node Pool will be placed. Changing this forces a new resource to be created.
```

**Review by tombuildsstuff — CHANGES_REQUESTED** [review]
hey @favoretti 

Thanks for this PR :)

Taking a look through this is looking good, if we can fix up the minor comments then this otherwise LGTM 👍 

Thanks!

File: azurerm/internal/services/containers/kubernetes_nodepool.go (hunk 2 above)
Comment (favoretti) [reply azurerm/internal/services/containers/kubernetes_nodepool.go:139]: Oh, I looked for it, but apparently overlooked. Apologies.

File: azurerm/internal/services/containers/kubernetes_nodepool.go (hunk 3 above)
Comment (favoretti) [reply azurerm/internal/services/containers/kubernetes_nodepool.go:138]: Ah, wasn't sure, docs/API wasn't stating this clearly. Let's make it so.

**Review by tombuildsstuff — APPROVED** [review]
LGTM - thanks for pushing those changes @favoretti

**Comment by tombuildsstuff** [comment]
@favoretti looks like the PPG ID test fails due to an invalid config?

```
------- Stdout: -------
=== RUN   TestAccAzureRMKubernetesClusterNodePool_proximityPlacementGroupId
=== PAUSE TestAccAzureRMKubernetesClusterNodePool_proximityPlacementGroupId
=== CONT  TestAccAzureRMKubernetesClusterNodePool_proximityPlacementGroupId
TestAccAzureRMKubernetesClusterNodePool_proximityPlacementGroupId: testing.go:684: Step 0 error: config is invalid: Unsupported argument: An argument named "proximity_placement_group_id" is not expected here.
--- FAIL: TestAccAzureRMKubernetesClusterNodePool_proximityPlacementGroupId (0.22s)
FAIL
```

**Comment by favoretti** [comment]
Ugh. I'll look into it in a bit and run acceptance here first too.

**Comment by favoretti** [comment]
```
$ TF_ACC=1 go test -v ./azurerm/internal/services/containers/tests/ -timeout=1000m -run TestAccAzureRMKubernetesClusterNodePool_proximityPlacementGroupId
=== RUN   TestAccAzureRMKubernetesClusterNodePool_proximityPlacementGroupId
=== PAUSE TestAccAzureRMKubernetesClusterNodePool_proximityPlacementGroupId
=== CONT  TestAccAzureRMKubernetesClusterNodePool_proximityPlacementGroupId
--- PASS: TestAccAzureRMKubernetesClusterNodePool_proximityPlacementGroupId (1184.41s)
PASS
ok  	github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/containers/tests	1185.558s
```

**Review by tombuildsstuff — APPROVED** [review]
LGTM - thanks for pushing those changes @favoretti

**Comment by tombuildsstuff** [comment]
Tests look good 👍

**Pull request merged**